### PR TITLE
[catch2] Update to version 2.13.7

### LINKED ISSUE #16938: [Catch2] update to 2.13.4
Library name: Catch2

New version number: 2.13.4

https://github.com/catchorg/Catch2

### PR DESCRIPTION
**Pull request by Ryan-rsm-McKenzie**
**Describe the pull request**

- #### What does your PR fix?  
  Updates Catch2 to 2.13.7. Also fixes #16938

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  Just an update

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  Yes

### CODE REVIEW

**Review by JonLiu1993 — APPROVED** [review]

File: ports/catch2/portfile.cmake [hunk 1]
```diff
@@ -1,27 +1,27 @@
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
-    REPO catchorg/Catch2
-    REF 5c88067bd339465513af4aec606bd2292f1b594a # v2.13.6
-    SHA512 62ab120ef9cbbcf7320a96654bda60c766dbbcc0d9cbb2b0b36dd04e828315b627caf51e390dcea915efa266655fe0f28058b972c0d6e0e3e457c565d26e1fd3
+    REPO "catchorg/Catch2"
+    REF "v2.13.7"
```
Comment (JonLiu1993) [inline ports/catch2/portfile.cmake:4]:
```suggestion
    REPO catchorg/Catch2
    REF v2.13.7
```

File: ports/catch2/portfile.cmake [hunk 2]
```diff
@@ -1,27 +1,27 @@
...
-vcpkg_configure_cmake(
-    SOURCE_PATH ${SOURCE_PATH}
-    PREFER_NINJA
+vcpkg_cmake_configure(
+    SOURCE_PATH "${SOURCE_PATH}"
     OPTIONS
-        -DBUILD_TESTING=OFF
-        -DCATCH_BUILD_EXAMPLES=OFF
+        "-DBUILD_TESTING=OFF"
+        "-DCATCH_BUILD_EXAMPLES=OFF"
```
Comment (JonLiu1993) [inline ports/catch2/portfile.cmake:13]:
```suggestion
        -DBUILD_TESTING=OFF
        -DCATCH_BUILD_EXAMPLES=OFF
```

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

**Review by JonLiu1993 — APPROVED** [review]

**Review by BillyONeal — APPROVED** [review]

**Comment by Ryan-rsm-McKenzie** [comment]
`rtaudio` failed because it attempted to link `libjack` when it wasn't present. It looks to me like this line should be checking `JACK_LIB_FOUND` instead of `JACK_LIB`:
https://github.com/thestk/rtaudio/blob/a68a68a8c4cdf79c2a5242e985a40becd79114c4/CMakeLists.txt#L16
Additionally, we should be passing `RTAUDIO_API_JACK=OFF` to `rtaudio`, since it's not listed in the port dependencies:
https://github.com/thestk/rtaudio/blob/a68a68a8c4cdf79c2a5242e985a40becd79114c4/CMakeLists.txt#L57

**Comment by BillyONeal** [comment]
I believe that was done by https://github.com/microsoft/vcpkg/pull/19516

**Review by JonLiu1993 — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks for your help!

**Pull request merged**